### ISSUE TITLE
User is seen as subscribed when the trial hasn't ended but the subscription was canceled

### ISSUE THREAD

**Issue by Livijn** [laravel/cashier-stripe#1551]
### Cashier Stripe Version

14.1

### Laravel Version

10.4

### PHP Version

8.1

### Description

As stated in the title, when a user has canceled (immediately) the subscription while on trial period, it is seen as subscribed. In my case here is the row that was affected:

| id | user_id | name    | stripe_id                    | stripe_status | stripe_price                   | quantity | trial_ends_at       | ends_at             | created_at          | updated_at          |
|----|---------|---------|------------------------------|---------------|--------------------------------|----------|---------------------|---------------------|---------------------|---------------------|
|  8 |       5 | profile | sub_XXX | canceled      | price_XXX |        1 | 2025-05-18 14:16:34 | 2023-07-11 08:07:18 | 2023-05-29 14:01:30 | 2023-07-11 08:07:18 |

Basically `ends_at` should take precedence over `trial_ends_at`.

### Steps To Reproduce

No need

**Comment by driesvints** [comment]
Thanks. This is actually fixed in the upcoming 15.x release: https://github.com/laravel/cashier-stripe/pull/1465

Right now we don't have an ETA, sorry.
